### ISSUE TITLE
Not escaped ampersand  used in attribute/prop creates errors in next lines

### ISSUE THREAD

**Issue by krystian-wolarek** [vuejs/eslint-plugin-vue#1918]
**Tell us about your environment**
- **ESLint version: 8.18.0** 
- **eslint-plugin-vue version: 9.1.1** 
- **Node version: 4.16.0** 
- **Operating System: macOS Monterey 12.3** 

**Please show your full configuration:** 
Clean Vue project with 1 additional rule

```json5
"rules": {
      "vue/max-attributes-per-line": [
        "error",
        {
          "singleline": {
            "max": 1
          },
          "multiline": {
            "max": 1
          }
        }
      ]
    }
```

**What did you do?**	

When we place any attribute ( prop or normal html attribute) that ends with `&` like in example bellow then eslint complains in next line about errors that not exist. See `What actually happened?`

When i add any letter after ampersand or escape it manually it working fine. See `What did you expect to happen?`

```vue
<some-component
    separator="&"
    any-attribute="Vue logo"
    class="anything" />
```

**What did you expect to happen?**
<img width="362" alt="Zrzut ekranu 2022-06-21 o 14 28 20" src="https://user-images.githubusercontent.com/39270768/174799066-e3cf06d0-f88a-4edb-9fc0-72b914dc8b80.png">

**What actually happened?**
<img width="748" alt="Zrzut ekranu 2022-06-21 o 14 27 28" src="https://user-images.githubusercontent.com/39270768/174798930-25edad02-bcc4-43aa-a065-eaa46eaecd86.png">

**Repository to reproduce this issue**

https://codesandbox.io/s/vigilant-chihiro-e3s7xz?file=/src/App.vue
